util/cmd/release/main.go

```diff
@@ -94,24 +93,143 @@ func main() {
 		// The windows binaries are suffixed with '.exe'. This allows us to create
 		// tarballs of the executables whether or not they contain a suffix.
 		files, _ := filepath.Glob(filepath.Join(dir, "protoc-gen-go_gapic*"))
-		util.Execute(
+		execute(
 			"tar",
 			"-zcf",
 			dir+".tar.gz",
 			"-C",
 			filepath.Dir(files[0]),
 			filepath.Base(files[0]))
 		// Remove the individual binary directory.
-		util.Execute("rm", "-r", dir)
+		execute("rm", "-r", dir)
 	}
 
 	// Execute GitHub release of artifacts.
-	util.Execute(
+	execute(
 		"ghr",
 		"-t="+token,
+		"-n="+version,
+		"-b='"+notes(version)+"'",
 		"-u=googleapis",
 		"-r=gapic-generator-go",
 		"-c="+commitish,
 		version,
 		stagingDir)
 }
+
+// execute runs the given strings as a command.
+func execute(cmd string, args ...string) {
+	if output, err := exec.Command(cmd, args...).CombinedOutput(); err != nil {
+		log.Fatalf("%s", output)
+	}
+}
+
+// executeWithOutput runs the given strings as a command and returns the output.
+func executeWithOutput(cmd string, args ...string) string {
+	out, err := exec.Command(cmd, args...).Output()
+	if err != nil {
+		log.Fatalf("%s", err)
+	}
+	return string(out)
+}
+
+// notes collects the commit messages since the previous tag, aggregates
+// dependency updates into one message, and returns formatted release notes.
+func notes(version string) string {
+	previous := executeWithOutput(
+		"git",
+		"describe",
+		"--abbrev=0",
+		"--tags",
+		version+"^",
+	)
+	previous = strings.TrimSpace(previous)
+
+	output := executeWithOutput(
+		"git",
+		"log",
+		previous+"..HEAD",
+		"--oneline",
+		"--pretty=format:%s",
+	)
+	commits := strings.Split(output, "\n")
+
+	return format(commits)
+}
+
+// format parses the given commits messages and formats them into release notes
+// organized by component.
+func format(commits []string) string {
+	var hasDeps bool
+	var gapic, bazel, gencli, chore, samples, other []string
```

Comment:
Sorry, I just thought of this, so feel free to ignore it.

What if we did a `map[string][]string`? Then we wouldn't need to duplicate all of the `append` and `WriteString` calls. Need to be careful with sorting and preventing other categories. So, it might end up being more complicated.

Reply:
That is a great idea, but I like the simplicity of the current approach so I think I'll stick with that :)